### ISSUE TITLE
[Feature]: Allow Obsidian Sync to sync the reminder states before reminding.

### ISSUE THREAD

**Issue by RobColes** [uphy/obsidian-reminder#85]
### Goal or desired outcome of this feature

I use Obsidian on 2 work PCs, one home and one phone. Obsidian Sync is set up on all of them. I would like to be able to check off the reminder once and not have it recur on the other devices when they are first connected, before they have had a chance to sync the new reminder state. 

### Describe the feature

For my use case, detecting completion of an initial Obsidian sync before starting the overdue checking process would be ideal, but I appreciate not everyone uses Obsidian Sync.

Maybe a configurable delay before the initial check?

**Comment by uphy** [comment]
I'm not sure because I'm not using Obsidian Sync, but I can implement the feature to delay the initial reminder check optionally by setting.

**Comment by RobColes** [comment]
That would be great, thank you. Happy to help beta test if that is useful.

**Comment by crhallberg** [comment]
I'm in the same boat here where I've had to disable Reminders for the time being since it keeps modifying files before they sync, creating conflicts.

**Comment by crhallberg** [comment]
I did some research on the Obsidian Discord and I found a post from a few months ago that outlines how to tap into the sync plugin (unofficially).

```js
let syncPlugin = app.internalPlugins.plugins["sync"].instance;

//Sync plugin turned on or off
console.log(syncPlugin.plugin.enabled);

//Sync plugin paused or not
console.log(syncPlugin.pause);

//Sync plugin currently in the middle of syncing
console.log(syncPlugin.syncing);
console.log(syncPlugin.getStatus());

//Hook into "status-change" event for a callback when syncing has started or has completed. This event fires alot while syncing so need to put criteria around it
syncPlugin.on("status-change", () => {
    console.log("sync status is", syncPlugin.syncStatus);
});

//Sync plugin status; "Fully synced" means good to go.
console.log(syncPlugin.syncStatus);
```

**Comment by ackatz** [comment]
Would be interested in this feature as well.

**Comment by Grant837** [comment]
Yes, I use obsidian on 4 devices, and its very annoying to have all the reminders pop up, that I just postponed for 3 hours, when I switch to my mobile device.

**Comment by Grant837** [comment]
@uphy > I'm not sure because I'm not using Obsidian Sync, but I can implement the feature to delay the initial reminder check optionally by setting it.

Alas, it has no impact - when you set it to 300 on all devices (and relaunch) and later start Obsidian on another device, the reminders pop up immediately

**Comment by Skroderider** [comment]
> Alas, it has no impact - when you set it to 300 on all devices (and relaunch) and later start Obsidian on another device, the reminders pop up immediately

I'm in the same boat as @Grant837. Is Reminders checking immediately after starting up? 
Am happy to get some logs for you if that will assist with troubleshooting.

**Comment by STP404** [comment]
I'm experiencing the same issue...

I use three computers and an Android phone, all synced using Resilio Sync (the vault is in a shared folder, and the app is installed normally).

Even after marking a reminder as completed, when I turn on any of the other machines, I get alerts for due reminders, which can be a lot if I haven't used that machine in a while.

A bigger issue is how I use special notes as recurring reminders. For example, a note might have a reminder like:

- [ ] 2025-07-01

When I get the notification, I open the note, do the task, and then update the date to something in the future, like:

- [ ] 2025-10-01

The problem is that when I start another machine, I still get all the alerts for outdated reminders. Even if I dismiss them, they continue to show up as "due" in the side panel. The plugin doesn't seem to recognize that the note itself has been updated and the old reminder no longer exists.

**But the worst part is this:
If I accidentally click "done" on one of these false reminders, it actually changes the note back to the old date, so:**

[ ] 2025-10-01 becomes [x] 2025-07-01

**This is a major issue and can seriously mess up task tracking.**
